connectors/sources/zoom.py

```diff
@@ -0,0 +1,540 @@
+                return await response.text()
+        except NotFound:
+            self._logger.warning(f"Content for {url} is being skipped.")
+
+    async def scroll(self, url):
+        scroll_url = url
+
+        while True:
+            response = await self.fetch(url=scroll_url)
+
+            if not response:
+                break
+
+            yield response
+
+            if next_page_token := response.get("next_page_token", None):
+                scroll_url = f"{url}&next_page_token={next_page_token}"
+            else:
+                break
+
+
+class ZoomClient:
+    def __init__(self, configuration):
+        self._sleeps = CancellableSleeps()
+        self._logger = logger
+
+        self.configuration = configuration
+        self.http_session = aiohttp.ClientSession(raise_for_status=True)
+
+        self.api_token = ZoomAPIToken(
+            http_session=self.http_session,
+            configuration=configuration,
+            logger_=self._logger,
+        )
+        self.api_client = ZoomAPISession(
+            http_session=self.http_session,
+            api_token=self.api_token,
+            logger_=self._logger,
+        )
+
+    def set_logger(self, logger_):
+        self._logger = logger_
+        self.api_token.set_logger(self._logger)
+        self.api_client.set_logger(self._logger)
+
+    async def close(self):
+        await self.http_session.close()
+        self.api_client.close()
+
+    async def get_users(self):
+        url = APIS["USERS"].format(base_url=BASE_URL, page_size=MEETING_PAGE_SIZE)
+        async for users in self.api_client.scroll(url=url):
+            for user in users.get("users", []) or []:
+                yield user
+
+    async def get_meetings(self, user_id, meeting_type):
+        url = APIS["MEETINGS"].format(
+            base_url=BASE_URL,
+            user_id=user_id,
+            meeting_type=meeting_type,
+            page_size=MEETING_PAGE_SIZE,
+        )
+        async for meetings in self.api_client.scroll(url=url):
+            for meeting in meetings.get("meetings", []) or []:
+                yield meeting
+
+    async def get_past_meeting(self, meeting_id):
+        url = APIS["PAST_MEETING"].format(base_url=BASE_URL, meeting_id=meeting_id)
+        return await self.api_client.fetch(url=url)
+
+    async def get_past_meeting_participants(self, meeting_id):
+        url = APIS["PAST_MEETING_PARTICIPANT"].format(
+            base_url=BASE_URL, meeting_id=meeting_id, page_size=MEETING_PAGE_SIZE
+        )
+        async for participants in self.api_client.scroll(url=url):
+            for participant in participants.get("participants", []) or []:
+                yield participant
+
+    async def get_recordings(self, user_id):
+        # Zoom recording does not retrieve data that is more than 4 months old.
+        now = datetime.utcnow()
+        for _ in range(4):
+            past_duration = now + timedelta(days=-30)
```

Comment:
"duration" usually means a delta. This variable is holding a specific date. Can we rename this variable and its usages to something like `start_date` and rename the `now` variable to be `end_date`?

IMO
```
end_date = start_date
```

is more logical as a loop iterates than:
```
now = past_duration
```